### DISCUSSION TITLE
Dynamic Google Calendar ID

### DISCUSSION THREAD

**Discussion by johneslabra** [spatie/laravel-google-calendar#296] [Q&A] [answered]
I'm trying to embed google calendar in our dashboard that has multiple users, is there's away that the dynamic calendar is Dynamic?

**Comment by johneslabra** [comment, marked as answer]
I found a workaround on this, for create event
go to "vendor\spatie\laravel-google-calendar\src\Event.php"
got to line 61 (Create Function)
add calendar_id as params
change $event->calendarId = static::getGoogleCalendar($calendarId)->getCalendarId();

to

$event->calendarId=$calendar_id;

in short

 public static function create(array $properties, string $calendarId = null, $optParams = [],$calendar_id=null)
    {
        $event = new static;

        // $event->calendarId = static::getGoogleCalendar($calendarId)->getCalendarId();
        $event->calendarId=$calendar_id;
        foreach ($properties as $name => $value) {
            $event->$name = $value;
        }

        return $event->save('insertEvent', $optParams);
    }

**Reply by chazkame** [reply]
Hi, I'm interested how to enable creating events using this package but to different Google Calendar Id depending on the needs. Can you please elaborate more?
Thank you

**Comment by donbalbieran** [comment]
Override the ENV by passing the 4th param, calendar id when you make the call to method Event

/** Override ENV default Leave Calendar with  */
    $college_calendar = 'xxxxxxxxxxxgroup.calendar.google.com';
    $googleEvents = Event::get(null,null,[],$college_calendar);
